### PR TITLE
rgw_sal_motr : [CORTX-30149] Fix multipart object issues in Delete Object API for unversioned bucket

### PR DESCRIPTION
**Pull request by priyanka25081999**
Problem Statement:
Delete object API is not working as expected for multipart upload object for unversioned bucket after adding null version-id code changes

Solution:
For unversioned bucket, Pass the key name as `objectname[version_id]` to the delete object API for multipart object.

Signed-off-by: Priyanka Salunke <priyanka.s.salunke@seagate.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -1729,7 +1729,7 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       {
         // delete object permanently.
         result.version_id = ent.key.instance;
-        if (source->category == RGWObjCategory::MultiMeta)
+        if (ent.meta.category == RGWObjCategory::MultiMeta)
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:1732]: why this change, source->category not reflecting proper one ? 
I can see ent.meta.category getting set in MotrMultipartUpload::init(), if this ent.meta.category reference getting used everywhere, then its fine.

Comment (priyanka25081999) [reply src/rgw/rgw_sal_motr.cc:1732]: Yes, `ent.meta.category` reference is used to store the object category during multipart object PUT. 
In the delete_obj() function, `source->category` returns None as it points to the MotrObject::MotrDeleteOp. In the same function, there is a call for get_bucket_dir_ent() function to fetch the object entry which gives us all the object data into ent structure. Then we can use the same to check the category by doing `ent.meta.category`

Comment (cdeshmukh) [reply src/rgw/rgw_sal_motr.cc:1732]: ok

**Review by cdeshmukh — APPROVED** [review]

**Review by sachitanands — APPROVED** [review]
looks good . 
these changes are enabling multipart deletion with un- versioned objects

**Review by shriya-deshmukh — APPROVED** [review]

**Pull request merged**